Zona update

kaitj opened the pull request:
## Proposed changes

Refactors the `zona_bb_subcortex.smk` file. Includes the following:

* Defines the zona output directory
* Changes for `snakefmt`
* Fix name of files in resources for zona
* Added anticipated path for anat segmentation for transformation purposes between different MNI spaces and subject native space
  * These files are available locally at the moment, but are too big to be added to the git repo (any ideas here @tkkuehn? I think hippunfold and maybe another workflow downloads any large files separately).
* Add transformation to rule to define an transform matrix

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [ ] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [x] Code has been run through the `poe quality` task
- [x] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

kaitj commented:
Okay, hopefully caught most things here. Should be ready for a review now.

tkkuehn commented on dbsc/workflow/rules/zona_bb_subcortex.smk:
```diff
@@ -1,234 +1,257 @@
+            datatype="anat",
+            space="T1w",
+            desc="ConvexHull",
+            suffix="segbin.nii.gz",
```
`mask`?

kaitj replied:
Do you mean to replace this as `mask.nii.gz`?

tkkuehn reviewed (changes requested):
A couple of things that need attention

kaitj commented on dbsc/workflow/rules/zona_bb_subcortex.smk:
```diff
@@ -1,234 +1,257 @@
+        config["singularity"]["neuroglia-core"]
     shell:
-        'applywarp --rel --interp=nn -i {input.mask} -r {input.ref} -w {input.xfm} -o {output.mask}'
+        "applywarp --rel --interp=nn -i {input.mask} -r {input.ref} -w {input.xfm} -o {output.mask}"
```
Nope, this is a bug and something that was left over from before - will look into fixing this.

kaitj commented on dbsc/workflow/rules/zona_bb_subcortex.smk:
```diff
@@ -1,234 +1,257 @@
-        'fslmaths {input.seg} -thr 15 {output.rm_seg} '
-        'fslmaths {input.seg} -sub {output.rm_seg} {output.seg}'
-        'fslmaths {output.seg} -add {output.non_thal} {output.seg}'
+        "fslmaths {input.seg} -sub 2 {output.non_thal} "
```
Yes, although I left everything within the block untouched since I figured labelmerge was going to replace this.

kaitj commented:
Also have to deal with this too:

```
* Added anticipated path for anat segmentation for transformation purposes between different MNI spaces and subject native space 
  * These files are available locally at the moment, but are too big to be added to the git repo (any ideas here @tkkuehn? I think hippunfold and maybe another workflow downloads any large files separately).
```

kaitj commented:
Updated to make use of `functools.partial`

Also removed the expand in the `xfm_zona_rois` rule, but need to double check to see if:
1) it grabs the correct inputs
2) it runs without error
